library/jdbc/spring-boot/forage-jdbc-starter/src/main/java/org/apache/camel/forage/springboot/jdbc/ForageDataSourceAutoConfiguration.java

```diff
@@ -162,6 +166,24 @@ private List<ServiceLoader.Provider<DataSourceProvider>> findDataSourceProviders
         return providers;
     }
 
+    private void createAggregationRepository(
+            ConfigurableBeanFactory configurableBeanFactory,
+            DataSourceFactoryConfig dsFactoryConfig,
+            DataSource agroalDataSource) {
+        if (!dsFactoryConfig.transactionEnabled() && dsFactoryConfig.aggregationRepositoryName() != null) {
+            log.warn("Transactions have to be enabled in order to create aggregation repositories");
+            return;
+        }
+        if (dsFactoryConfig.aggregationRepositoryName() != null) {
+            configurableBeanFactory.registerSingleton(
+                    dsFactoryConfig.aggregationRepositoryName(),
+                    new ForageAggregationRepository(
+                            agroalDataSource,
+                            com.arjuna.ats.jta.TransactionManager.transactionManager(),
+                            dsFactoryConfig));
```

Comment:
Same note about having this as a interface (i.e.: separating behavior from implementation). I think could give us some flexibility in the future.

Reply:
Totally, I am going to open another PR for the idempotent repository that follows similar patterns, and there's room for improvement